server/utils/match/occupancyCalendar.test.ts

```diff
@@ -87,6 +87,25 @@ describe('occupancyCalendar', () => {
     ])
   })
 
+  it.each([
+    ['foo/:date', 'foo/2025-02-12'],
+    [':date', '2025-02-12'],
+    ['foo/:date?query', 'foo/2025-02-12?query'],
+    ['foo?date=:date', 'foo?date=2025-02-12'],
```

Comment:
This is probably overkill, as  we most likely won't need to replace other parts than path parameters, but it serves to show the intent, that is, the substring `:date` is replaced in any kind of placeholder URL.